flutter/lib/device_info.dart

```diff
@@ -5,24 +5,31 @@ import 'package:flutter/services.dart';
 import 'package:device_info_plus/device_info_plus.dart';
 import 'package:device_marketing_names/device_marketing_names.dart';
 import 'package:mlperfbench_common/data/environment/environment_info.dart';
+import 'package:process_run/shell.dart';
+
+import 'package:mlperfbench/backend/bridge/ffi_cpuinfo.dart';
 
 class DeviceInfo {
   final String? modelCode;
   final String? modelName;
   final String? manufacturer;
+  final EnvAndroidInfo? androidInfo;
```

Comment:
On iOS and Windows cpuinfo is able to provide CPU name.
On Android we often get `Unknown` instead of the CPU/SoC name when querying cpuinfo lib, so we try to gather some additional android-specific info.

Reply:
Can we restructure or integrate that info into other field or class? It's quite strange that there is only one `androidInfo` and not other `iosInfo`, `windowsInfo`.

Reply:
I think it makes sense to put data that's only available on one OS into one field, and set this field to null when the OS is different.
If we mode data from `EnvAndroidInfo` into some other place, we will still have "this field is present on Android, null on other OS", except we will have several such fields.

Reply:
This is mainly used to display `Device SoC/CPU name`? If yes, can we just merge / concatenate the `EnvAndroidInfo` into `EnvCpuinfo.socName`? Else there should be another `iosInfo` or `windowsInfo` because they certainly also have SoC info like model or manufacturer.

Reply:
`EnvAndroidInfo` does not contain actual cpu info.
It's some additional info that can be used to decode cpu info when it's missing.

>can we just merge / concatenate the EnvAndroidInfo into EnvCpuinfo

We can create fields:
`EnvCpuinfo.android_prop_soc_model`: nullable, only present on android, doesn't make sense on ios/windows.
`EnvCpuinfo.android_prop_soc_manufacturer`: nullable, only present on android, doesn't make sense on ios/windows.
`EnvCpuinfo.android_build_board`: nullable, only present on android, doesn't make sense on ios/windows.

I think it would be more confusing than putting these android-specific fields into a separate map.

>Else there should be another iosInfo or windowsInfo because they certainly also have SoC info like model or manufacturer

If we encounter any ios/windows devices that are missing `EnvCpuinfo.socName` field, we may have to create os-specific fields for them.
Right now theoretical `iosInfo` and `windowsInfo` fields are not needed because SoC model+manufacturer are contained in the `EnvCpuinfo.socName` field.

Reply:
> 1. Why is the `EnvCpuinfo.socName` not a field of `EnvSocInfo`?

Because `EnvCpuinfo.socName` is data from `cpuinfo` lib.

> 2\. Why is there only `EnvAndroidInfo` but not for iOS, macOS or Windows (as the original question)?

Because `EnvCpuinfo.socName` is different on different operating systems.
On iOS and Windows it always contain valid data.
On Android there is  50/50 chance that `EnvCpuinfo.socName` is valid, so we have to add os-specific fields to make CPU detection more consistent.

Reply:
> The structure is inconsistent.

The structure is: `environmentInfo map` -> `socInfo map` -> `data source type` -> `actual data`.

Reply:
I suggest using this structure to have the data unified.
```dart
enum EnvSocInfoDataSourceEnum { cpuinfo, getprop, board, ... }
class EnvSocInfo {
  final String socName;
  final EnvSocInfoDataSourceEnum dataSource;
}
```

Reply:
What would be the algorithm to ensure that `socName` is valid and consistent?

iOS and Windows are OK since on these systems we have single reliable source of data.

On Android currently we use the following algorithm:
If board has a known value, use board.
If getprop returned valid data, use it.
Else fall back to cpuinfo.

The problem here is that `board` value is mapped to a proper name via a database that may be temporary outdated.
So we can't decode board in current version of the app but we will be able to do it in later versions.

This is why I kept all of the source data. If we have all the data, we may potentially be able to show something readable even if we encounter unknown board value.

Reply:
The main question is what to do with `socName`. If it's only used for UI display, just make `socName` nullable and display what it has.
If you want to keep all the data as detailed as possible, then I suggest using this structure:
```dart
socInfo: EnvSocInfo(
  iosInfo: EnvSocInfoIos(
    socName: 'soc name',
    iosOnlyField: 'ios only data'
  ),
  windowsInfo: EnvSocInfoWindows(
    socName: 'soc name'
  ),
  androidInfo: EnvSocInfoAndroid(
    socName: 'soc name'
    propSocModel: 'soc model',
    propSocManufacturer: 'soc manufacturer',
    buildBoard: 'board'
  ),
),
```
The point is that all fields in the same level of hierarchy should have something in common. I don't think that the choice of underlying 3rd party libs should impact the data structure.

Reply:
I changed environment info data layout to have device-specific fields for each OS.
Except It made more sense to me to also separate model name and stuff.